op-node/p2p/store/ip_ban_book.go

```diff
@@ -84,6 +80,9 @@ func (d *ipBanBook) SetIPBanExpiration(ip net.IP, expirationTime time.Time) erro
 	if expirationTime == (time.Time{}) {
 		return d.book.deleteRecord(ip.To16().String())
 	}
+	// Should these move up to include the delete?
+	d.book.Lock()
+	defer d.book.Unlock()
```

Comment:
I think that makes sense, as deleting a write operation.

Reply:
I double checked. Because store and cache are both self-synchronized, it's "safe". However it's also possible to end up caching a value right after deleting it. I think putting the lock around this better overall.